httpclient5/src/main/java/org/apache/hc/client5/http/socket/PlainConnectionSocketFactory.java

```diff
@@ -63,7 +63,7 @@ public PlainConnectionSocketFactory() {
 
     @Override
     public Socket createSocket(final Proxy proxy, final HttpContext context) throws IOException {
-        return proxy != null ? new Socket(proxy) : new Socket();
```

Comment:
@Sineaggi This looks wrong. The check for the proxy being non null should be performed by the method.

Reply:
Then perhaps this code should be

```
return proxy != null ? new Socket(proxy) : createSocket(context);
```

As it stands, any overridden implementation of `createSocket(HttpContext)` is ignored, which is a breaking change.

Reply:
In the interface, yes, but that delegation is not reflected in PlainConnectionSocketFactory's implementation of the method. This class is not final and thus users can extend it, overriding createSocket. If that is incorrect usage and this class was meant to be final, I think that should at least be documented in the class's javadoc.

Reply:
@twarner-sofi While `PlainConnectionSocketFactory` is non final and can be extended, one _must_ be aware of risks of tight coupling through inheritance and risks of subclassing a concrete class and overriding methods that were not meant for extension in the first place. Do not pin this problem on us.

Reply:
Okay, thanks for confirming that these methods are not meant for extension in the first place. It's unfortunate that they weren't marked `final`, or documented as such, but that's fine now.

We'll look into making a fix PR for docker-java.

Reply:
> one must be aware of risks of tight coupling through inheritance and risks of subclassing a concrete class and overriding methods that were not meant for extension in the first place

The class in question has no private fields and no hidden internal logic, and has existed virtually unmodified since 2012. Without warning, this public and non-final class has had a breaking change made to its behavior in a patch release of this library.

Reply:
@Sineaggi There is no breaking change. The class remains fully API compatible. Any particular modality or implementation details have never been a part of the public contract. You abused class inheritance in your code.